Support group concept in the alarm core (#5615)
apache/skywalking: docs/en/setup/backend/backend-alarm.md

backend deployment env time.
- **Count**. In the period window, if the number of **value**s over threshold(by OP), reaches count, alarm
should send.

> - **Only as condition**. Specify if the rule can send notfiy or just as an conditon of composite rule

Author: done

- **Silence period**. After alarm is triggered in Time-N, then keep silence in the **TN -> TN + period**.
By default, it is as same as **Period**, which means in a period, same alarm(same ID in same 
metrics name) will be trigger once. 

> ### Composite rules

Reviewer: ```suggestion
### Composite rules
**NOTE**. Composite rules only work for alarm rules targeting the same entity level, such as alarm rules of the service level. For example, `service_percent_rule && service_resp_time_percentile_rule`. You shouldn't composite alarm rules of different entities, such as one alarm rule of the service metrics with another rule of the endpoint metrics.

```

Author: done

Composite rule is constituted by following keys
- **Rule name**. Unique name, show in alarm message. Must end with `_rule`.

> - **Expression**. Specify how to compose rules, support and/or expression

Reviewer: Please test the `parenthesis` case, this is important with `and` and `or`.

Author: added test case

backend deployment env time.
- **Count**. In the period window, if the number of **value**s over threshold(by OP), reaches count, alarm
should send.
- **Only as condition**. Specify if the rule can send notification or just as an condition of composite rule.
- **Silence period**. After alarm is triggered in Time-N, then keep silence in the **TN -> TN + period**.
By default, it is as same as **Period**, which means in a period, same alarm(same ID in same 
metrics name) will be trigger once. 

### Composite rules
**NOTE**. Composite rules only work for alarm rules targeting the same entity level, such as alarm rules of the service level. 

> For example, `service_percent_rule && service_resp_time_percentile_rule`. You shouldn't compose alarm rules of different entities, 

Author: Yes, I knew that, is it better to support different entities?

> For example, `service_percent_rule && service_resp_time_percentile_rule`. You shouldn't compose alarm rules of different entities, 

Reviewer: > Yes, I knew that, is it better to support different entities?

I am wondering what is the case of that. I was making the words accurate.

> For example, `service_percent_rule && service_resp_time_percentile_rule`. You shouldn't compose alarm rules of different entities, 

Author: Never mind, please ignore this

such as one alarm rule of the service metrics with another rule of the endpoint metrics.

> Composite rule is constituted by following keys

Reviewer: ```suggestion

Composite rule is constituted by the following keys
```

Fix format issue.

    count: 3
    # How many times of checks, the alarm keeps silence after alarm triggered, default as same as period.
    silence-period: 10

>     # Specify if the rule can send notfiy or just as an conditon of composite rule

Author: done

    count: 3
    silence-period: 5
    message: The request number of entity {name} non-200 status is more than expected.
    only-as-condition: false
composite-rules:
  comp_rule:
    # Must satisfied percent rule and resp time rule 
    expression: service_percent_rule && service_resp_time_percentile_rule
    message: Service percent less than 80% and response time p50 > 1000
```

### Default alarm rules
We provided a default `alarm-setting.yml` in our distribution only for convenience, which including following rules
1. Service average response time over 1s in last 3 minutes.
2. Service success rate lower than 80% in last 2 minutes.
3. Percentile of service response time is over 1s in last 3 minutes
4. Service Instance average response time over 1s in last 2 minutes, and the instance name matches the regex.
5. Endpoint average response time over 1s in last 2 minutes.
6. Database access average response time over 1s in last 2 minutes.

> 7. Endpoint relation average response time over 1s in last 2 minutes.

Reviewer: ```suggestion
1. Service success rate lower than 80% in last 2 minutes.
1. Percentile of service response time is over 1s in last 3 minutes
1. Service Instance average response time over 1s in last 2 minutes, and the instance name matches the regex.
1. Endpoint average response time over 1s in last 2 minutes.
1. Database access average response time over 1s in last 2 minutes.
1. Endpoint relation average response time over 1s in last 2 minutes.
```

This is markdown, you don't need to increase by yourself.

    count: 3
    silence-period: 5
    message: The request number of entity {name} non-200 status is more than expected.
    only-as-condition: false
composite-rules:
  comp_rule:
    # Must satisfied percent rule and resp time rule 
    expression: service_percent_rule && service_resp_time_percentile_rule

>     message: Service percent less than 80% and response time p50 > 1000

Reviewer: If so, please add a test case for this too, besides change the documents.

>     message: Service percent less than 80% and response time p50 > 1000

Author: Currently, it will not support message format, will add it later

>     message: Service percent less than 80% and response time p50 > 1000

Author: done
